Basket feature

christopher-timm opened the pull request:
This adds a basket functionality.

Attention: I was not able to build the OL3 library successfully. Some OL3 dependencies might be missing.

Update: I've merged the master branch into my feature branch. 
Please check the correctness of the file tx_dlf_pageview.js, because of merge conflicts

christopher-timm committed:
add basket plugin

jacmendt commented on dlf/plugins/pageview/tx_dlf_ol3.js:
```diff
@@ -107,6 +107,10 @@ ol.Map.prototype.rotate = function(rotation) {
         rotate = view.getRotation() + (rotation *  Math.PI/180),
         center = view.getCenter();
 
+    if (tx_dlf_viewer.ov_view != null) {
```
This introduces a recursion in the code. When the `ov_view` objects exists it calls the prototype method `rotate` of the `ol.Map` object, which leads again to this expression. It is better to bind the map views via openlayers events with each other. Also see the `adjustViews` method in the image manipulation control.

``` javascript
var adjustViews = function(sourceView, destMap) {
            var rotateDiff = sourceView.getRotation() !== destMap.getView().getRotation();
            var resDiff = sourceView.getResolution() !== destMap.getView().getResolution();
            var centerDiff = sourceView.getCenter() !== destMap.getView().getCenter();

            if (rotateDiff || resDiff || centerDiff) {
                destMap.zoomTo(sourceView.getCenter(),
                    sourceView.getZoom(), 50);
                destMap.getView().rotate(sourceView.getRotation());
            }

        }
```

jacmendt commented on dlf/plugins/pageview/tx_dlf_pageview.js:
```diff
@@ -396,4 +444,215 @@ dlfViewer.prototype.initLayer = function(imageSourceObjs, isCorsEnabled) {
+
+/**
+ * @constructor
+ * @extends {ol.interaction.Pointer}
```
Take care that this also works with the openlayer custom build. Add `ol.interaction.Pointer` and all other _new_ openlayers dependencies to [build config file](https://github.com/chrizzor/goobi-presentation/blob/107682ec9daa4810aa9d3969fdcb9ffda87a6984/dlf/lib/OpenLayers/ol3-dlf.json). The current build is based on the [slub fork of openlayers - branch: kitodo-presentation](https://github.com/slub/ol3/tree/kitodo-presentation).

christopher-timm replied:
[ca6fb3a]

jacmendt commented on dlf/plugins/pageview/tx_dlf_pageview.js:
```diff
@@ -396,4 +444,215 @@ dlfViewer.prototype.initLayer = function(imageSourceObjs, isCorsEnabled) {
+    var dlfViewer = this;
+
+    this.map.on('pointermove', function (evt) {
+        mousePosition = dlfViewer.map.getEventCoordinate(evt.originalEvent);
```
See the comment on openlayers custom builds.

christopher-timm replied:
[ca6fb3a]

jacmendt commented on dlf/plugins/pageview/tx_dlf_pageview.js:
```diff
@@ -396,4 +444,215 @@ dlfViewer.prototype.initLayer = function(imageSourceObjs, isCorsEnabled) {
+
+dlfViewer.prototype.setNewCropDrawer = function  (source) {
+	viewerObject = this;
+	this.draw = new ol.interaction.Draw({
```
See the comment on openlayers custom builds.

jacmendt commented on dlf/plugins/pageview/tx_dlf_pageview.js:
```diff
@@ -396,4 +444,215 @@ dlfViewer.prototype.initLayer = function(imageSourceObjs, isCorsEnabled) {
+        var extent = geometry.getExtent();
+        var imageExtent = viewerObject.map.getLayers().item(0).getExtent();
+
+        var pixel = ol.extent.getIntersection(imageExtent, extent);
```
See the comment on openlayers custom builds.

christopher-timm replied:
[ca6fb3a]

jacmendt commented on dlf/plugins/pageview/tx_dlf_pageview.js:
```diff
@@ -396,4 +444,215 @@ dlfViewer.prototype.initLayer = function(imageSourceObjs, isCorsEnabled) {
+ */
+dlfViewer.prototype.activateSelection = function () {
+	this.resetCropSelection();
+    this.map.addInteraction(this.draw);
```
See the comment on openlayers custom builds.

christopher-timm replied:
[ca6fb3a]

jacmendt commented on dlf/plugins/pageview/tx_dlf_pageview.js:
```diff
@@ -396,4 +444,215 @@ dlfViewer.prototype.initLayer = function(imageSourceObjs, isCorsEnabled) {
       });
 
     return deferredResponse;
-};
```
This will break the code.

christopher-timm replied:
[d5081d1]

jacmendt commented on dlf/plugins/pageview/tx_dlf_pageview.js:
```diff
@@ -366,6 +401,19 @@ dlfViewer.prototype.init = function(controlNames) {
 
+    this.source = new ol.source.Vector();
+    // crop selection style
+    this.selection = new ol.interaction.DragBox({
```
See the comment on openlayers custom builds.

jacmendt commented on dlf/plugins/pageview/tx_dlf_pageview.js:
```diff
@@ -396,4 +444,215 @@ dlfViewer.prototype.initLayer = function(imageSourceObjs, isCorsEnabled) {
+        rotation: rotation,
+    });
+
+    this.ov_map = new ol.Map({
```
Does the function needs an own map object? For preventing to further blow the `tx_dlf_pageview.js` file, you could also think about seperating your code into an own javascript file.

christopher-timm replied:
It needs an own map object, because the zoom level between origin map and the magnifier map should be different as well as the magnifier map should not have any control functionality.

sebastian-meyer commented on dlf/ext_tables.php:
```diff
@@ -217,9 +217,108 @@
+		'default_sortby' => 'ORDER BY label',
+		'delete'	=> 'deleted',
+		'dynamicConfigFile' => \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::extPath($_EXTKEY).'tca.php',
+		'iconfile'	=> \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::extPath($_EXTKEY).'res/icons/txdlflibraries.png',
```
Please choose a different icon which better represents the purpose of this table.
The icon set can be found here: http://www.famfamfam.com/lab/icons/silk/

christopher-timm replied:
[411e449]

sebastian-meyer commented on dlf/ext_tables.php:
```diff
@@ -217,9 +217,108 @@
+		'default_sortby' => 'ORDER BY label',
+		'delete'	=> 'deleted',
+		'dynamicConfigFile' => \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::extPath($_EXTKEY).'tca.php',
+		'iconfile'	=> \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::extPath($_EXTKEY).'res/icons/txdlflibraries.png',
```
Please choose a different icon which better represents the purpose of this table.
The icon set can be found here: http://www.famfamfam.com/lab/icons/silk/

christopher-timm replied:
[411e449]

sebastian-meyer commented on dlf/ext_tables.php:
```diff
@@ -217,9 +217,108 @@
+		'default_sortby' => 'ORDER BY label',
+		'delete'	=> 'deleted',
+		'dynamicConfigFile' => \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::extPath($_EXTKEY).'tca.php',
+		'iconfile'	=> \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::extRelPath($_EXTKEY).'res/icons/txdlflibraries.png',
```
Please choose a different icon which better represents the purpose of this table.
The icon set can be found here: http://www.famfamfam.com/lab/icons/silk/

christopher-timm replied:
[411e449]

sebastian-meyer commented on dlf/locallang.xml:
```diff
@@ -133,10 +144,21 @@
+			<label index="tx_dlf_printer.label">Label</label>
+			<label index="tx_dlf_printer.address">Networkaddress</label>
+			<label index="tx_dlf_printer.name">Name</label>
+			<label index="tx_dlf_printer.printcommand">Drucker Cli Befehl (##fileName##)</label>
```
This label should be in English. :o)

christopher-timm replied:
Should be fixed [2857c31]

sebastian-meyer commented on dlf/locallang.xml:
```diff
@@ -133,10 +144,21 @@
 			<label index="tx_dlf_toolbox.toolsFulltext">Fulltext</label>
 			<label index="tx_dlf_toolbox.toolsImagemanipulation">Image Manipulation</label>
 			<label index="tt_content.dlf_audioplayer">DLF: Audio Player</label>
+			<label index="tx_dlf_toolbox.toolsPublicationDownload">Publication Download</label>
```
Please remove the `toolsPublicationDownload` label.

christopher-timm replied:
Label has been removed [431822a]

sebastian-meyer commented on dlf/locallang.xml:
```diff
@@ -315,10 +348,21 @@
 			<label index="tx_dlf_toolbox.toolsFulltext">Volltext</label>
 			<label index="tx_dlf_toolbox.toolsImagemanipulation">Bildbearbeitung</label>
 			<label index="tt_content.dlf_audioplayer">DLF: Audioplayer</label>
+			<label index="tx_dlf_toolbox.toolsPublicationDownload">Publication Download</label>
```
Please remove the `toolsPublicationDownload` label.

christopher-timm replied:
[431822a]

sebastian-meyer commented on dlf/plugins/basket/class.tx_dlf_basket.php:
```diff
@@ -0,0 +1,763 @@
+		$this->setCache(FALSE);
+
+		// Load the list.
+		// $this->list = t3lib_div::makeInstance('tx_dlf_basket');
```
Please remove inactive code.

christopher-timm replied:
[5b79304]

sebastian-meyer commented on dlf/plugins/basket/class.tx_dlf_basket.php:
```diff
@@ -0,0 +1,763 @@
+		// 	if (isset($this->piVars['selected'])) {
+		// 		$this->printDocument();
+		// 	}
+		// }
```
Please remove inactive code.

christopher-timm replied:
[5b79304]

sebastian-meyer commented on dlf/plugins/basket/class.tx_dlf_basket.php:
```diff
@@ -0,0 +1,763 @@
+			'parameter' => $this->conf['targetBasket'],
+			'additionalParams' => t3lib_div::implodeArrayForUrl($this->prefixId, $additionalParams, '', TRUE, FALSE)
+		);
+		// $link = $this->cObj->typoLink_URL('Go to basket', $conf);
```
Please remove inactive code as the generated link is not used anywhere.

christopher-timm replied:
[5b79304] and [d57eb24]

sebastian-meyer commented on dlf/plugins/basket/class.tx_dlf_basket.php:
```diff
@@ -0,0 +1,763 @@
+
+			$basketData['doc_ids'] = $items;
+
+			// return array('basketData' => $basketData, 'jsOutput' => $output);
```
Please remove inactive code.

christopher-timm replied:
[5b79304]

sebastian-meyer commented on dlf/plugins/basket/class.tx_dlf_basket.php:
```diff
@@ -0,0 +1,763 @@
+			// 	// $pageNums = $document->numPages;
+			// } else {
+			// 	$pageNums = 1;
+			// }
```
Please remove inactive code.

christopher-timm replied:
[5b79304]

sebastian-meyer commented on dlf/plugins/basket/class.tx_dlf_basket.php:
```diff
@@ -0,0 +1,763 @@
+
+		// printer is selected
+		if ($printerData) {
+			// $pdfUrl = $this->conf['pdfprint'];
```
Please remove inactive code.

christopher-timm replied:
[5b79304]

sebastian-meyer commented on dlf/plugins/basket/locallang.xml:
```diff
@@ -0,0 +1,87 @@
+      <label index="chooseMail">E-Mail Empfänger auswählen</label>
+		</languageKey>
+	</data>
+</T3locallang>
```
Please remove unused entries from `locallang` file.

christopher-timm replied:
[26b9ae7]

sebastian-meyer commented on dlf/plugins/interaction/class.tx_dlf_interaction.php:
```diff
@@ -0,0 +1,133 @@
+ */
+
+/**
+ * Plugin 'DLF: List View' for the 'dlf' extension.
```
This should say `DLF: Interaction`

christopher-timm replied:
[ee1511d]

sebastian-meyer commented on dlf/plugins/interaction/class.tx_dlf_interaction.php:
```diff
@@ -0,0 +1,133 @@
+            });';
+            $output .= '</script>';
+
+            $markerArray['###BASKETBUTTON###'] = $output;//$this->cObj->typoLink($label, $basketConf);//'<a href="">add to basket</a>';
```
Please remove inactive code.

christopher-timm replied:
[ee1511d]

sebastian-meyer commented on dlf/plugins/interaction/flexform.xml:
```diff
@@ -0,0 +1,117 @@
+        <sDEF>
+            <ROOT>
+                <TCEforms>
+                    <sheetTitle>LLL:EXT:dlf/plugins/navigation/locallang.xml:tt_content.pi_flexform.sheet_general</sheetTitle>
```
Please fix the path: it should say `interaction` instead of `navigation`.

christopher-timm replied:
[b79f538]

sebastian-meyer commented on dlf/plugins/interaction/flexform.xml:
```diff
@@ -0,0 +1,117 @@
+                    <targetPid>
+                        <TCEforms>
+                            <exclude>1</exclude>
+                            <label>LLL:EXT:dlf/plugins/navigation/locallang.xml:tt_content.pi_flexform.targetPid</label>
```
Please fix the path: it should say `interaction` instead of `navigation`.

christopher-timm replied:
[b79f538]

sebastian-meyer commented on dlf/plugins/interaction/flexform.xml:
```diff
@@ -0,0 +1,117 @@
+                    <templateFile>
+                        <TCEforms>
+                            <exclude>1</exclude>
+                            <label>LLL:EXT:dlf/plugins/navigation/locallang.xml:tt_content.pi_flexform.templateFile</label>
```
Please fix the path: it should say `interaction` instead of `navigation`.

christopher-timm replied:
[b79f538]

sebastian-meyer commented on dlf/plugins/interaction/locallang.xml:
```diff
@@ -0,0 +1,51 @@
+<T3locallang>
+    <meta type="array">
+        <type>module</type>
+        <description>Language labels for plugin tx_dlf_navigation</description>
```
This is the language file for `tx_dlf_interaction`, not `tx_dlf_navigation`.

christopher-timm replied:
[2df8c34]

sebastian-meyer commented on dlf/plugins/interaction/template.tmpl:
```diff
@@ -0,0 +1,26 @@
+-->
+<!-- ###TEMPLATE### -->
+<div class="tx-dlf-navigation-addbasket">###BASKETBUTTON###</div>
+<div class="tx-dlf-navigation-basket">###BASKET###</div>
```
The element's classes should be `tx-dlf-interaction-...`.

christopher-timm replied:
[2df8c34]

sebastian-meyer commented on dlf/plugins/listview/class.tx_dlf_listview.php:
```diff
@@ -281,6 +281,21 @@ protected function getEntry($number, $template) {
+		// basket button
+		$markerArray['###BASKETBUTTON###'] = '';
+		if (!empty($this->conf['basketButton']) && !empty($this->conf['targetBasket'])) {
+			// $additionalParams = array ('id' => $this->list[$number]['uid'], 'addToBasket' => true);
```
Please remove inactive code.

christopher-timm replied:
[48099f1]

sebastian-meyer commented on dlf/plugins/listview/class.tx_dlf_listview.php:
```diff
@@ -281,6 +281,21 @@ protected function getEntry($number, $template) {
+		if (!empty($this->conf['basketButton']) && !empty($this->conf['targetBasket'])) {
+			// $additionalParams = array ('id' => $this->list[$number]['uid'], 'addToBasket' => true);
+			$additionalParams = array ('id' => $this->list[$number]['uid'], 'startpage' => $this->list[$number]['page'], 'addToBasket' => 'list');
+			// $additionalParams = array ('id' => 1, 'addToBasket' => 'list');
```
Please remove inactive code.

christopher-timm replied:
[48099f1]

sebastian-meyer commented on dlf/plugins/listview/class.tx_dlf_listview.php:
```diff
@@ -495,7 +510,22 @@ protected function getSubEntries($number, $template) {
+			// basket button
+			$markerArray['###SUBBASKETBUTTON###'] = '';
+			if (!empty($this->conf['basketButton']) && !empty($this->conf['targetBasket'])) {
+				// $additionalParams = array ('id' => $this->list[$number]['uid'], 'addToBasket' => true);
```
Please remove inactive code.

christopher-timm replied:
[48099f1]

sebastian-meyer commented on dlf/plugins/listview/class.tx_dlf_listview.php:
```diff
@@ -495,7 +510,22 @@ protected function getSubEntries($number, $template) {
+			if (!empty($this->conf['basketButton']) && !empty($this->conf['targetBasket'])) {
+				// $additionalParams = array ('id' => $this->list[$number]['uid'], 'addToBasket' => true);
+				$additionalParams = array ('id' => $this->list[$number]['uid'], 'startpage' => $subpart['page'], 'endpage' => $subpart['page'], 'logId' => $subpart['sid'], 'addToBasket' => 'subentry');
+				// $additionalParams = array ('id' => 1, 'addToBasket' => 'list');
```
Please remove inactive code.

christopher-timm replied:
[48099f1]

sebastian-meyer commented on dlf/plugins/pageview/template.tmpl:
```diff
@@ -24,5 +24,11 @@
 <div id="tx-dlf-map" class="tx-dlf-map">
 	<div id="tx-dlf-fulltextselection"></div>
 </div>
+<div id="ov_map"></div>
```
Please remove if separate map object isn't needed (see comments on javascript file).

sebastian-meyer commented on dlf/plugins/toc/class.tx_dlf_toc.php:
```diff
@@ -84,18 +84,30 @@ protected function getMenuEntry(array $entry, $recursive = FALSE) {
 			$entryArray['doNotLinkIt'] = 0;
 
+			if ($this->conf['basketButton']) {
+				$entryArray['basketButtonHref'] = '<a href="'.$this->pi_linkTP_keepPIvars_url(array ('addToBasket' => 'toc', 'logId' => $entry['id'], 'startpage' => $entry['points']), TRUE, FALSE, $this->conf['targetBasket']).'">'.$this->pi_getLL('basketButton', '', TRUE).'</a>';//'<a href="basketButtonLink">BasketButtonTODOTRANSLATION</a>';
```
Please remove inactive code.

christopher-timm replied:
[5417a2d]

sebastian-meyer commented on dlf/plugins/toc/class.tx_dlf_toc.php:
```diff
@@ -84,18 +84,30 @@ protected function getMenuEntry(array $entry, $recursive = FALSE) {
 			$entryArray['doNotLinkIt'] = 0;
 
+			if ($this->conf['basketButton']) {
+				$entryArray['basketButtonHref'] = '<a href="'.$this->pi_linkTP_keepPIvars_url(array ('addToBasket' => 'toc', 'logId' => $entry['id'], 'startpage' => $entry['points']), TRUE, FALSE, $this->conf['targetBasket']).'">'.$this->pi_getLL('basketButton', '', TRUE).'</a>';//'<a href="basketButtonLink">BasketButtonTODOTRANSLATION</a>';
```
Please remove inactive code.

christopher-timm replied:
[5417a2d]

sebastian-meyer commented on dlf/plugins/toc/class.tx_dlf_toc.php:
```diff
@@ -84,18 +84,30 @@ protected function getMenuEntry(array $entry, $recursive = FALSE) {
 			$entryArray['doNotLinkIt'] = 0;
 
+			if ($this->conf['basketButton']) {
+				$entryArray['basketButtonHref'] = '<a href="'.$this->pi_linkTP_keepPIvars_url(array ('addToBasket' => 'toc', 'logId' => $entry['id'], 'startpage' => $entry['targetUid']), TRUE, FALSE, $this->conf['targetBasket']).'">'.$this->pi_getLL('basketButton', '', TRUE).'</a>';//'<a href="basketButtonLink">BasketButtonTODOTRANSLATION</a>';
```
Please remove inactive code.

christopher-timm replied:
[5417a2d]

sebastian-meyer commented on dlf/plugins/toc/flexform.xml:
```diff
@@ -57,6 +57,30 @@
+					<basketButton>
+						<TCEforms>
+							<exclude>1</exclude>
+							<label>LLL:EXT:dlf/plugins/listview/locallang.xml:tt_content.pi_flexform.basketButton</label>
```
Please fix the path: it should say `toc` instead of `listview`.

christopher-timm replied:
[3afa808]

sebastian-meyer commented on dlf/plugins/toc/flexform.xml:
```diff
@@ -57,6 +57,30 @@
+					<targetBasket>
+						<TCEforms>
+							<exclude>1</exclude>
+							<label>LLL:EXT:dlf/plugins/listview/locallang.xml:tt_content.pi_flexform.targetBasket</label>
```
Please fix the path: it should say `toc` instead of `listview`.

christopher-timm replied:
[3afa808]

sebastian-meyer commented on dlf/plugins/toc/setup.txt:
```diff
@@ -19,6 +19,8 @@ plugin.tx_dlf_toc.menuConf {
 				}
 				dataWrap = <span class="tx-dlf-toc-title">|</span> <span class="tx-dlf-toc-pagination">{field:pagination}</span>
 			}
+			allWrap = |<span class="tx-dlf-basket-button">{field:basketButtonHref}</span>
```
Shouldn't this wrap only be executed, if `field:basketButton` is set?

sebastian-meyer commented on dlf/tca.php:
```diff
@@ -1181,3 +1181,186 @@
+				'eval' => 'required,trim',
+			),
+		),
+		'index_name' => array (
```
There is no field `index_name` in the table `tx_dlf_basket`.

sebastian-meyer commented on dlf/tca.php:
```diff
@@ -1181,3 +1181,186 @@
+				'eval' => 'alphanum,unique',
+			),
+		),
+		'type_name' => array (
```
There is no field `type_name` in the table `tx_dlf_basket`.

sebastian-meyer commented on dlf/tca.php:
```diff
@@ -1181,3 +1181,186 @@
+				'max' => 255,
+				'eval' => 'alphanum,unique',
+			),
+		),
```
There are some database fields missing here.

sebastian-meyer commented:
In addition to the line notes some general notes:
- Please change all the license headers from `Goobi. Digitalisieren im Verein` to `Kitodo. Key to digital objects`
- Please have a look at the TCA control fields and check your code accordingly: https://docs.typo3.org/typo3cms/TCAReference/Reference/Ctrl/Index.html
- In general there is a mismatch between the database tables defined in `ext_tables.sql`, the TCA control array in `ext_tables.php` and the full TCA configuration in `tca.php`.
- There are a lot of copy&paste errors in your code. Cloning code templates from other plugins is perfectly OK, but you should make sure that all path references and labels are changed accordingly.
- Please remove inactive or unused code.

stweil commented:
I suggest to squash the commit history of this PR a little bit - especially commits which were later reverted. Or do we need the whole development history in the official git? I don't think so.

stweil commented:
IMHO this PR has far too many commits and really needs squashing. There is no use in commits which add syntax errors which are fixed or even reverted in later commits. If we ever want to be able to use git bisect, then it must be possible to choose any intermediate revision without getting broken code.

And a side note: do we still have to support PHP < 5.5? I don't think so, as PHP 5.5 is already "old stable".

christopher-timm commented:
Do i have to squash my commits? Or is it the responsability for the one who is merging this feature to the master branch?

stweil commented:
You know your commits better than anybody else, so you are the person who should do the squashing.

Don't confuse squashing commits of a pull request with the action offered by GitHub during the merge ("squash and merge"). Squashing your commits means using "git rebase -i", combining related commits, bringing the commits in a reasonable order and finally pushing the result (which normally needs "git push -f").

christopher-timm committed:
add basket and add basket buttons

christopher-timm committed:
add interaction plugin

christopher-timm committed:
pageview cropping

christopher-timm committed:
db and form structure

christopher-timm committed:
basket feature: cleanup + bugfixing

christopher-timm committed:
add icons

christopher-timm commented:
Ok, the commits are squashed.
- [x] Merge master into feature_branch
- [x] Check functionality

Feature is ready for merge

christopher-timm committed:
Merge branch 'remote_master' into z_feature

# Conflicts:
#	dlf/ext_autoload.php
#	dlf/ext_localconf.php
#	dlf/ext_tables.php
#	dlf/lib/OpenLayers/ol3-dlf.json
#	dlf/locallang.xml
#	dlf/plugins/listview/class.tx_dlf_listview.php
#	dlf/plugins/pageview/class.tx_dlf_pageview.php
#	dlf/plugins/pageview/template.tmpl
#	dlf/plugins/pageview/tx_dlf_pageview.js

christopher-timm committed:
merge conflict cleanup

sebastian-meyer committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into pr/117

sebastian-meyer committed:
Code cleanup and conflict resolving

sebastian-meyer commented:
After resolving the conflicts and cleaning up the code I merged your branch into `master` and pushed the result to [pr/117](https://github.com/kitodo/kitodo-presentation/tree/pr/117). Please have a look at the code and verify that everything is working as expected. :)

christopher-timm committed:
move functionality from interaction plugin to pageview and basket plugin

christopher-timm committed:
remove interaction plugin

christopher-timm committed:
add intval for user data input

christopher-timm committed:
code cleanup

christopher-timm committed:
basket fix document title

sebastian-meyer committed:
Merge branch 'z_feature' of https://github.com/chrizzor/goobi-presentation into chrizzor-z_feature

sebastian-meyer committed:
Merge branch 'chrizzor-z_feature' into pr/117

sebastian-meyer commented:
@chrizzor I merged your pull request into master and pushed the result to the [pr/117](https://github.com/kitodo/kitodo-presentation/tree/pr/117) branch. Please check again if everything is working as expected.

Also have a look at the pageview plugin. You seem to have implemented a rotation feature for OpenLayers although there already is one. Please make sure that there is no unnecessary duplication of features. (You have re-added a lot of OL libraries to dlf/lib/OpenLayers/ol3-dlf.json. Are those necessary or was this an unintended regression?)

christopher-timm committed:
remove rotation feature because it already exists

christopher-timm committed:
add rotation synchronization for the magnifier map

christopher-timm commented:
@sebastian-meyer I removed the rotation feature and added the rotation synchronization for the magnifier view in the existing rotation function.
The OL libraries are necessary for the magnifier and crop function.

sebastian-meyer commented:
Looks good now! Now please update your branch to base on current master and we are ready to merge!

christopher-timm committed:
Merge branch 'remote_master' into z_feature

# Conflicts:
#	dlf/ext_tables.php
#	dlf/plugins/navigation/locallang.xml
#	dlf/plugins/pageview/flexform.xml
#	dlf/plugins/pageview/locallang.xml
#	dlf/plugins/pageview/tx_dlf_pageview.js
#	dlf/tca.php

sebastian-meyer commented:
@chrizzor Unfortunately I had to revert the merge-commit of the basket feature, because it breaks a lot of other things:

- Merging instead of re-basing your branch reverted a lot of work from `master`
- Your pull request re-introduces old `TCA` and `ext_tables` configuration which was already rewritten to be compatible with TYPO3 7.x
- At some places you use old API calls which are no longer supported and break TYPO3 7.x compatibility

Please fix those problems and create a new pull request. I am sorry. :(
